packages/core/src/envelope.ts

```diff
@@ -68,7 +68,10 @@ export function createEventEnvelope(
   tunnel?: string,
 ): EventEnvelope {
   const sdkInfo = getSdkMetadataForEnvelopeHeader(metadata);
-  const eventType = event.type || 'event';
+
+  // TODO: Handle replay_event here
+  // Currently, this is done by replay, but we want to upstream this here
+  const eventType = event.type && event.type !== 'replay_event' ? event.type : 'event';
```

Comment:
To note, this is technically correct, but may increase the bundle size slightly for "nothing", as this will not really be hit right now. I'd still rather leave this in and remove it once we (hopefully soon!) actually handle replay events "normally". WDYT?

Reply:
Sounds good to me.